clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/ingestion/heartbeat/HeartbeatMonitoringService.java

```diff
@@ -74,36 +76,32 @@ private synchronized void initializeEntry(
     if (version.getHybridStoreConfig() == null) {
       return;
     }
-    if (heartbeatTimestamps.get(version.getStoreName()) == null) {
-      heartbeatTimestamps.put(version.getStoreName(), new VeniceConcurrentHashMap<>());
-    }
-    if (heartbeatTimestamps.get(version.getStoreName()).get(version.getNumber()) == null) {
-      heartbeatTimestamps.get(version.getStoreName()).put(version.getNumber(), new VeniceConcurrentHashMap<>());
-    }
-    if (heartbeatTimestamps.get(version.getStoreName()).get(version.getNumber()).get(partition) == null) {
-      Map<String, Long> regionTimestamps = new VeniceConcurrentHashMap<>();
-      if (version.isActiveActiveReplicationEnabled()) {
-        for (String region: regionNames) {
-          regionTimestamps.put(region, DEFAULT_SENTINEL_HEARTBEAT_TIMESTAMP);
-        }
-      } else {
-        regionTimestamps.put(localRegionName, DEFAULT_SENTINEL_HEARTBEAT_TIMESTAMP);
-      }
-      heartbeatTimestamps.get(version.getStoreName()).get(version.getNumber()).put(partition, regionTimestamps);
-    }
+    heartbeatTimestamps.computeIfAbsent(version.getStoreName(), storeKey -> new VeniceConcurrentHashMap<>())
+        .computeIfAbsent(version.getNumber(), versionKey -> new VeniceConcurrentHashMap<>())
+        .computeIfAbsent(partition, partitionKey -> {
+          Map<String, Long> regionTimestamps = new VeniceConcurrentHashMap<>();
+          if (version.isActiveActiveReplicationEnabled()) {
+            for (String region: regionNames) {
+              regionTimestamps.put(region, DEFAULT_SENTINEL_HEARTBEAT_TIMESTAMP);
+            }
+          } else {
+            regionTimestamps.put(localRegionName, DEFAULT_SENTINEL_HEARTBEAT_TIMESTAMP);
+          }
+          return regionTimestamps;
+        });
   }
 
   private synchronized void removeEntry(
       Map<String, Map<Integer, Map<Integer, Map<String, Long>>>> heartbeatTimestamps,
       Version version,
       int partition) {
-    if (heartbeatTimestamps.get(version.getStoreName()) != null) {
-      if (heartbeatTimestamps.get(version.getStoreName()).get(version.getNumber()) != null) {
-        if (heartbeatTimestamps.get(version.getStoreName()).get(version.getNumber()).get(partition) != null) {
-          heartbeatTimestamps.get(version.getStoreName()).get(version.getNumber()).remove(partition);
-        }
-      }
-    }
+    heartbeatTimestamps.computeIfPresent(version.getStoreName(), (storeKey, map1) -> {
```

Comment:
nit: can we call them `perVersionMap`, `perPartitionMap` etc. like you did for line 190? Prefer to not have them as `map1`, `map2`

Reply:
fixed